gammapy/modeling/models/spatial.py

```diff
@@ -1506,10 +1506,14 @@ def write(self, overwrite=False):
         overwrite: bool, optional
             Overwrite existing file.
             Default is False, which will raise a warning if the template file exists already.
+        filename: str, optional
+            Set the filename of the template  model.
```

Comment:
Maybe specify that this is an added convenience, and by default the template will be written to `self.filename`